gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -233,114 +413,57 @@ def run_translation_averaging(
-        # Possibly perform (slow!) outlier rejection with Minimum Feedback Arc Set algorithm.
-        if self._reject_outliers:
-            inliers: Set[Tuple[int, int]] = self.__compute_inlier_mask(w_i2Ui1_measurements)
-    
-            w_i2Ui1_inlier_measurements = BinaryMeasurementsUnit3()
-            for idx in range(len(w_i2Ui1_measurements)):
-                # key1 is i2 and key2 is i1 above.
-                w_i2Ui1 = w_i2Ui1_measurements[idx]
-                i1 = w_i2Ui1.key2()
-                i2 = w_i2Ui1.key1()
-                if (i1, i2) in inliers:
-                    w_i2Ui1_inlier_measurements.append(w_i2Ui1)
-        else:
-            inliers = set(i2Ui1_dict.keys())
+        w_i2Ui1_dict, valid_cameras = get_valid_measurements_in_world_frame(i2Ui1_dict, wRi_list)
 
-        # Run the optimizer
-        # TODO(akshay-krishnan): remove once latest gtsam pip wheels updated.
-        try:
-            algorithm = TranslationRecovery()
-            w_i2ti1_priors = self._get_prior_measurements_in_world_frame(i2Ti1_priors, wRi_list)
-            wti_initial = self.__get_initial_values(absolute_pose_priors)
-            if len(w_i2ti1_priors) > 0:
-                # scale is ignored here.
-                wti_values = algorithm.run(w_i2Ui1_inlier_measurements, 0.0, w_i2ti1_priors, wti_initial)
+        if self._use_tracks_for_averaging:
+            if tracks_2d is None:
+                logger.info("No tracks provided for translation averaging. Falling back to camera unit translations.")
+                w_i2Ui1_dict_tracks = {}
+            elif intrinsics is None or len(intrinsics) != len(wRi_list):
+                raise ValueError("Number of intrinsics must match number of rotations")
             else:
-                wti_values = algorithm.run(w_i2Ui1_inlier_measurements, scale_factor)
-        except TypeError as e:
-            logger.error("TypeError: {}".format(str(e)))
-            recovery = TranslationRecovery(w_i2Ui1_inlier_measurements)
-            wti_values = recovery.run(scale_factor)
+                selected_tracks = self._select_tracks_for_averaging(tracks_2d, valid_cameras, intrinsics)
+                w_i2Ui1_dict_tracks = self._get_landmark_directions(selected_tracks, intrinsics, wRi_list)
+        else:
+            w_i2Ui1_dict_tracks = {}
 
-        # Transform the result to a list of Point3.
-        wti_list = [
-            wti_values.atPoint3(i) if wRi_list[i] is not None and wti_values.exists(i) else None
-            for i in range(num_images)
-        ]
-        # Combine estimated global rotations and global translations to Pose(3) objects.
-        wTi_list = [
-            Pose3(wRi, wti) if wRi is not None and wti is not None else None for wRi, wti in zip(wRi_list, wti_list)
-        ]
+        w_i2Ui1_dict_inliers, w_i2Ui1_dict_tracks_inliers, inlier_cameras = self.compute_inliers(
+            w_i2Ui1_dict, w_i2Ui1_dict_tracks
+        )
+
+        wti_list = self.__run_averaging(
```

Comment:
nit: keyword args would be nice here if possible because we've got a lot of args